smartsim/_core/_install/builder.py

```diff
@@ -854,9 +860,19 @@ def supported_platforms() -> t.Sequence[t.Tuple[OperatingSystem, Architecture]]:
     def __rai_dependency_name__(self) -> str:
         return f"libtorch@{self.url}"
 
+    @staticmethod
+    def _patch_out_mkl(libtorch_root: Path) -> None:
+        _modify_source_files(
+            libtorch_root / "share/cmake/Caffe2/public/mkl.cmake",
+            r"find_package\(MKL QUIET\)",
+            "# find_package(MKL QUIET)",
+        )
+
     def __place_for_rai__(self, target: t.Union[str, "os.PathLike[str]"]) -> Path:
         self.extract(target)
         target = Path(target) / "libtorch"
+        if not self.with_mkl:
+            self._patch_out_mkl(target)
```

Comment:
Should we do this patch out on every extract, not just on extraction for RAI, sense `with_mkl` is a class attr?
```py
def extract(target):
    super().extract(target)
    if not self.with_mkl:
        self._patch_out_mkl(target / "libtorch")
```
?